### PR TITLE
Improve linter

### PR DESCRIPTION
**Pull request by paroxp**
What
----

I've noticed that some tests are not actually working, due to bugs in code.

We'd like to be more rigorous about the code to prevent certain problems
and help with debugging. Moving away from TSLint has given too much room
for error.

Some plugins come in helpful. For instance the a11y plugin, will help us
remember about accessibility issues when writing or refactoring the
code.

Most of these changes are adding spacing, reordering lines, making sure
the async await has been used properly.

There are however few exceptions...

Some codebase has been simplified and fix not to throw compiler errors.

Some tests have been "straightened" in a way, that proved the tests have
never been running or working properly.

Some codebase have been fixed, as the tests have FINALLY highlighted it
being broken.

How to review
-------------

- Deploy to your environment (or checkout mine)
- Click around and attempt to use crucial functionality (permissions for instance)

### CODE REVIEW

File: src/components/platform-admin/views.tsx [hunk 1]
```diff
@@ -95,7 +95,7 @@ function Costs(props: IFormProperties): ReactElement {
         </div>
 
         <div className="govuk-form-group">
-          <label className="govuk-label govuk-date-input__label">Format</label>
+          <label className="govuk-label govuk-date-input__label" htmlFor="view-costs-format">Format</label>
```
Comment (kr8n3r) [inline src/components/platform-admin/views.tsx]: 👏

File: src/components/reports/views.tsx [hunk 2]
```diff
@@ -460,7 +462,7 @@ export function CostByServiceReport(props: ICostByServiceReport): ReactElement {
         <tbody className="govuk-table__body">
           {props.billablesByOrganisationAndService.map(record => (
             <tr key={record.orgGUID} className="govuk-table__row">
-              <td className="govuk-table__cell" scope="row">
```
Comment (kr8n3r) [inline src/components/reports/views.tsx]: this needs to be kept

File: src/components/reports/views.tsx [hunk 3]
```diff
@@ -518,7 +520,7 @@ export function CostByServiceReport(props: ICostByServiceReport): ReactElement {
         <tbody className="govuk-table__body">
           {props.billablesByOrganisationAndSpaceAndService.map(record => (
             <tr key={record.spaceGUID} className="govuk-table__row">
-              <td className="govuk-table__cell" scope="row">
```
Comment (kr8n3r) [inline src/components/reports/views.tsx]: keep

File: src/components/spaces/views.tsx [hunk 4]
```diff
@@ -277,7 +274,7 @@ export function SpacesPage(props: ISpacesPageProperties): ReactElement {
         {props.organization.entity.name}.
       </p>
 
-      <details className="govuk-details" role="group">
```
Comment (kr8n3r) [inline src/components/spaces/views.tsx]: keep https://github.com/alphagov/govuk-frontend/blob/b57ade99ac353a944620d818c1f89e1005935181/src/govuk/components/date-input/template.njk#L80

File: src/layouts/template.tsx [hunk 5]
```diff
@@ -139,9 +140,10 @@ export class Template {
   private EnableClientSideJavaScript(): ReactElement {
     return (
       <script
+        // eslint-disable-next-line react/no-danger
         dangerouslySetInnerHTML={{
           __html:
-            `document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');`,
+            'document.body.className = ((document.body.className) ? document.body.className + \' js-enabled\' : \'js-enabled\');',
```
Comment (kr8n3r) [inline src/layouts/template.tsx]: hmmm, we should check that classes are correct when this is built

Comment (paroxp) [reply src/layouts/template.tsx]: I agree. I seem to recall this may be problematic. This is the linter auto-fixing things.

Comment (paroxp) [reply src/layouts/template.tsx]: Tested in my environment and the `js-enabled` has been correctly applied 🎉

File: src/components/spaces/views.tsx (hunk 4 above)
Comment (paroxp) [reply src/components/spaces/views.tsx]: Reverted back in separate commit, with an exception to the linter.

**Pull request merged**